infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/ShellScriptProvider.java

```diff
@@ -81,9 +81,13 @@ public InfrastructureProvisionResult provision(TestPlan testPlan)
             inputParameters.setProperty(OUTPUT_DIR, workspace);
             String parameterString = TestGridUtil.getParameterString(null, inputParameters);
             ShellExecutor executor = new ShellExecutor(null);
-            executor.executeCommand("bash " + Paths
-                    .get(testPlanLocation, createScript.getFile()) + " " + parameterString);
             InfrastructureProvisionResult result = new InfrastructureProvisionResult();
+            if (!executor.executeCommand("bash " + Paths
```

Comment:
Instead of checking boolean true/false, will it be better to throw error from executor.executeCommand function and catch it here?

Reply:
@Kasun: In that case, the caller of the ShellExecutor have to possess the knowledge of shell return statuses. For example, it has to know non-zero means a failure. IMO, we should handle all the shell script related logic inside the ShellExecutor and return a standard outcome to the caller.

Reply:
+1

Reply:
+1. We do not log the exit code anywhere as well.

Reply:
Changes made in 4991909